sharktank/tests/models/llama/benchmark_tests.py

```diff
@@ -0,0 +1,779 @@
+        Raises Exception if compilation fails for some reason.
+        """
+        cmd = self.get_compile_cmd(mlir_path, output_file, args)
+        logging.getLogger().info(f"Launching compile command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, capture_output=True, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to compile.")
+
+    def iree_benchmark_module(
+        self,
+        hip_device_id: str,
+        vmfb_name: str,
+        irpa_path: str,
+        args: List[str],
+        cwd: str | Path,
+    ):
+        """Runs a compiled program with the given args using `iree-benchmark-module`.
+        This assumes that the `iree-benchmark-module` command is available (usually via PATH).
+        Args:
+            vmfb_name: Name of the .vmfb file (relative to `cwd`).
+            args: List of arguments to pass to `iree-benchmark-module`.
+            cwd: Working directory to run the command within. (either string or Path works)
+            compile_cmd: Command used to compile the program, for inclusion in error messages.
+        Raises Exception if running fails for some reason.
+        """
+        benchmark_args = [
+            f"ROCR_VISIBLE_DEVICES={hip_device_id}",
+            "iree-benchmark-module",
+            f"--device=hip://{hip_device_id}",
+            "--hip_use_streams=true",
+            "--hip_allow_inline_execution=true",
+            "--device_allocator=caching",
+            f"--module={vmfb_name}",
+            f"--parameters=model={irpa_path}",
+        ]
+        benchmark_args += args
+        cmd = subprocess.list2cmdline(benchmark_args)
+        logging.getLogger().info(f"Launching run command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, stdout=sys.stdout, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to run")
+
+    def cleanup_output_files(
+        self, output_mlir_path: str, output_json_path: str, output_file: str
+    ):
+        try:
+            # Removing output_mlir_path
+            subprocess.run(f"rm {output_mlir_path}", shell=True, check=True)
+            print(f"Removed: {output_mlir_path}")
+
+            # Removing output_json_path
+            subprocess.run(f"rm {output_json_path}", shell=True, check=True)
+            print(f"Removed: {output_json_path}")
+
+            # Removing output_file
+            subprocess.run(f"rm {output_file}", shell=True, check=True)
+            print(f"Removed: {output_file}")
+
+        except subprocess.CalledProcessError as e:
+            print(f"Error removing files: {e}")
+
+
+class BenchmarkLlama3_1_8B(BaseBenchmarkTest):
+    def setUp(self):
+        # TODO: add numpy files to Azure and download from it
+        self.repo_root = os.getenv("SHARK_PLATFORM_REPO_ROOT")
+        artifacts_dir = "/data/extra/models/llama3.1_8B/"
```

Comment:
likely need at least a comment that this is system specific for A30F. Anybody trying to replicate locally will need to know how to recreate. Better would probably be adding an arg to pass this path that defaults to this for the runner

Reply:
General workflow tip for tests / CI: start by making tests runnable by a developer, _then_ teach a CI machine how to follow those steps. We should never have a hardcoded path like this (and this will only work on systems that support such a file path, so no Windows).